hw4/chris901003/_matrix.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <vector>
+#include <cstdint>
+
+using namespace std;
+
+template<typename T>
+class CustomAllocator: public std::allocator<T> {
+    public:
+    using value_type = T;
+    using pointer = T*;
+    using const_pointer = const T*;
+    using size_type = size_t;
+    
+    template<typename U>
+    struct rebind {
+        typedef CustomAllocator<U> other;
+    };

```

Comment:
[FYI](https://github.com/CnTransGroup/EffectiveModernCppChinese/blob/master/src/3.MovingToModernCpp/item9.md), you can replace this part with alias declarations.
```c++
template<typename U>
using rebind = CustomAllocator<U>;
```